### PR TITLE
Clean up librustc_typeck error_codes file

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
r? @Dylan-DPC

### CODE REVIEW

File: src/librustc_typeck/error_codes.rs [hunk 1]
```diff
@@ -185,11 +172,9 @@ match d {
 "##,
 
 E0029: r##"
-In a match expression, only numbers and characters can be matched against a
-range. This is because the compiler checks that the range is non-empty at
-compile-time, and is unable to evaluate arbitrary comparison functions. If you
-want to capture values of an orderable type between two end-points, you can use
-a guard.
+Something else than numbers and characters has been used for a range.
```
Comment (Mark-Simulacrum) [inline src/librustc_typeck/error_codes.rs]:
```suggestion
Something other than numbers and characters has been used for a range.
```

**Comment by Mark-Simulacrum** [comment]
I am personally feeling rather against these changes. They seem to move the main description down below the code example, which would at least for me be harmful rather than helpful (the code example isn't usually all that helpful, as I expect it to be essentially the code I've already written!). As such, I would personally not want to r+ this PR. I'm not sure -- but I suspect it'd be good to codify some guidelines before we go through all of these files and change them around. cc @rust-lang/wg-diagnostics -- seems like you'd be interested and would want to get involved.

**Comment by GuillaumeGomez** [comment]
We did make such rules a few years ago. The structure is supposed to be as follows:

```text
[short description]

Erroneous code example(s):

[failing code example]

[error explanation]

[fixed code example]
```

**Comment by Mark-Simulacrum** [comment]
Do you have a pointer to somewhere where that's documented? If not, would you be willing to make a PR to the forge? I personally think the previously existing "short descriptions" were better than the one-sentence examples this modifies them to -- and would consider them to be pretty short.

**Comment by GuillaumeGomez** [comment]
That's why I added @steveklabnik: I don't remember if such resource exists or not. If it does, we need to make it more easily accessible.

**Comment by JohnCSimon** [comment]
Ping from triage - this PR has sat idle for a week. 
@steveklabnik  can you comment on this?
cc: @GuillaumeGomez @Dylan-DPC @Mark-Simulacrum 

Thanks

**Comment by GuillaumeGomez** [comment]
Actually, it has a merged RFC: https://github.com/rust-lang/rfcs/pull/1567

So I guess it's good to go? Unless if we want to change the RFC?

**Comment by Mark-Simulacrum** [comment]
@bors r+ rollup

Thanks for citing the RFC! It'd be good to perhaps add comments to the top of all the error code files linking to the RFC (http://rust-lang.github.io/rfcs/1567-long-error-codes-explanation-normalization.html)?

**Comment by Mark-Simulacrum** [comment]
@bors r-

Actually, it'd be great to squash this first.

**Comment by GuillaumeGomez** [comment]
Squashed!

**Pull request merged**